CefSharp/Internals/JavascriptObjectRepository.cs

```diff
@@ -234,18 +237,85 @@ private void AnalyseObjectForBinding(JavascriptObject obj, bool analyseMethods,
                     var jsObject = CreateJavascriptObject(camelCaseJavascriptNames);
                     jsObject.Name = propertyInfo.Name;
                     jsObject.JavascriptName = GetJavascriptName(propertyInfo.Name, camelCaseJavascriptNames);
-                    jsObject.Value = jsProperty.GetValue(obj.Value);
+                    jsObject.LateBinding = () =>
+                    {
+                        var newValue = jsProperty.GetValue(obj.Value);
+                        if (newValue == jsObject.Value)
+                        {
+                            return;
+                        }
+                        SetJavascriptObjectValue(jsObject, newValue, propertyInfo.PropertyType, camelCaseJavascriptNames);
+                    };
                     jsProperty.JsObject = jsObject;
+                }
+                obj.Properties.Add(jsProperty);
+            }
+        }
 
-                    AnalyseObjectForBinding(jsProperty.JsObject, analyseMethods, readPropertyValue, camelCaseJavascriptNames);
+        private void SetJavascriptObjectValue(JavascriptObject jsObject, object value, Type type, bool camelCaseJavascriptNames)
+        {
+            if (type.IsGenericType)
+            {
+                IList list = value as IList;
+                if (list != null)
```

Comment:
Simplify this, evaluate `null` and throw exception straight away, reduces nesting.